File: ruby/test/sorts/insertion_sort_test.rb
require_relative '../../src/sorts/insertion_sort'
require_relative '../../src/util/util'
require_relative '../../src/util/constants'
require_relative '../util'

require 'test_helper'
require 'minitest/autorun'
require 'minitest/spec'

describe InsertionSort do
    describe 'when passed an array' do
        it 'should correctly sort it' do
            arr = TestUtil.random_array(Constants.ARRAY_SIZE)

            result = InsertionSort.sort(arr)

            Util.is_sorted(result).must_equal true
        end
    end
end

File: Java/test/sorts/InsertionSortTest.java
package sorts;

import org.junit.Assert;
import org.junit.Test;
import util.SortUtil;
import util.TestUtil;

/**
 * Created by Chris Keyser on 10/3/2016.
 */
public class InsertionSortTest {
    @Test
    public void InsertionSort_SortsArray() {
        int arr[] = TestUtil.generateRandomArray(TestUtil.ARR_SIZE);

        int[] sorted = InsertionSort.sort(arr);

        Assert.assertTrue(SortUtil.isSorted(sorted));
    }
}
